src/main/java/com/getindata/connectors/http/internal/sink/httpclient/JavaNetSinkHttpClient.java

```diff
@@ -43,50 +30,28 @@
 @Slf4j
 public class JavaNetSinkHttpClient implements SinkHttpClient {
 
-    private static final int HTTP_CLIENT_THREAD_POOL_SIZE = 16;
-
-    public static final String DEFAULT_REQUEST_TIMEOUT_SECONDS = "30";
-
-    private final HttpClient httpClient;
-
     private final String[] headersAndValues;
 
     private final Map<String, String> headerMap;
 
     private final HttpStatusCodeChecker statusCodeChecker;
 
-    private final HttpPostRequestCallback<HttpSinkRequestEntry> httpPostRequestCallback;
+    private final HttpPostRequestCallback<HttpRequest> httpPostRequestCallback;
 
-    /**
-     * Thread pool to handle HTTP response from HTTP client.
-     */
-    private final ExecutorService publishingThreadPool;
-
-    private final int httpRequestTimeOutSeconds;
-
-    public JavaNetSinkHttpClient(Properties properties, HeaderPreprocessor headerPreprocessor) {
-        this(properties, new Slf4jHttpPostRequestCallback(), headerPreprocessor);
-    }
+    private final RequestSubmitter requestSubmitter;
 
     public JavaNetSinkHttpClient(
-        Properties properties,
-        HttpPostRequestCallback<HttpSinkRequestEntry> httpPostRequestCallback,
-        HeaderPreprocessor headerPreprocessor) {
-
-        ExecutorService httpClientExecutor =
-            Executors.newFixedThreadPool(
-                HTTP_CLIENT_THREAD_POOL_SIZE,
-                new ExecutorThreadFactory(
-                    "http-sink-client-request-worker", ThreadUtils.LOGGING_EXCEPTION_HANDLER));
+            Properties properties,
+            HttpPostRequestCallback<HttpRequest> httpPostRequestCallback,
+            HeaderPreprocessor headerPreprocessor,
+            RequestSubmitterFactory requestSubmitterFactory) {
 
-        this.httpClient = JavaNetHttpClientFactory.createClient(properties, httpClientExecutor);
         this.httpPostRequestCallback = httpPostRequestCallback;
```

Comment:
refactoring moved to BatchRequestSubmitterFactory and SingleRequestSubmitterFactory